app.py

```diff
@@ -75,112 +94,264 @@ def toDict(responses, baseQuestions=None):
             responseAsDict[partie_avant] = partie_apres
     return responseAsDict
 
+# Function to read CSV file
 def read_csv_file(file_path):
-    # Lire le fichier CSV
+    # Read csv file
     df = pd.read_csv(file_path, sep=';')
     return df
 
-def  create_final_request(file, baseQuestions, typeOfQuestion):
+# Function to create base request
+def create_base_request(file):
+    requestAsText = ""
+    for index, line in file.iterrows():
+        if not line.isnull().all():
+            specification = line['Specification']
+            requestAsText += str(specification) + "\n"
+
+        line_text = str(line['Key']) + ":" + str(line['Question']) + "; \n"
+        requestAsText += line_text
+    return requestAsText
+
+# Function to create final request
+def  create_final_request(file, baseQuestions):
```

Comment:
There is significant code duplication in the functions create_final_request. It would be preferable to refactor this code by creating reusable functions.

Reply:
gonna be done in the next pr because this one is to big for now